faucet/main.go

```diff
@@ -221,20 +241,49 @@ func execMain(cfg *rootCfg) error {
 	// Create the client (HTTP)
 	cli := tm2Client.NewClient(cfg.remote)
 
-	// Prepare the middlewares
-	var middlewares []faucet.Middleware
+	// Prepare the faucet values
+	var (
+		middlewares []faucet.Middleware
+		prepareTxFn faucet.PrepareTxMessageFn
+	)
 
 	if len(cfg.allowedTokens) != 0 {
-		// TODO temporary for testing purpose without redis
+		// TODO temporary for testing purposes without redis
 		middlewares = append(middlewares, prepareTokenListMiddleware(cfg.allowedTokens))
+
+		// By default, for testing purposes, the transaction is a MsgSend
+		prepareTxFn = func(cfg faucet.PrepareCfg) std.Msg {
+			return bank.MsgSend{
+				FromAddress: cfg.FromAddress,
+				ToAddress:   cfg.ToAddress,
+				Amount:      cfg.SendAmount,
+			}
+		}
```

Comment:
Considering that `bank.MsgSend` is already the default behavior of the faucet when no `prepareTxFn` is provided, I don't think it's required to fill `prepareTxFn` here.

Reply:
Oh I've just looked at the faucet code, and if you provide a `prepareTxFn=nil`, it will probably panic when transferring funds.
The proper way is so to deal with a `[]faucet.Option` rather than a  `prepareTxFn` directly. When `allowedTokens` is not empty, you add the `faucet.WithPrepareTxFn` option, if it's empty you don't.